File: src/main/java/dev/entite/forum/Topic.java
package dev.entite.forum;

import com.fasterxml.jackson.annotation.JsonIgnore;
import dev.entite.BaseEntite;
import dev.entite.Utilisateur;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;
/**
 * Classe pour la définition des topics <br/>
 * Se référer à {@link dev.entite.BaseEntite} pour les identifiants générés
 */
@Entity
public class Topic extends BaseEntite {
    /**
     * Le titre du topic
     */
    private String libelle;
    /**
     * Relation many to one avec les utilisateurs <br/>
     * Id dans table topic = utilisateur_id <br/>
     * Voir {@link dev.entite.Utilisateur}
     */
    @ManyToOne
    private Utilisateur utilisateur;
    /**
     * Relation one to many avec les posts associés à un topic <br/>
     * Id dans table post = topic_id <br/>
     * Voir {@link dev.entite.forum.Post}
     */
    @OneToMany(mappedBy = "topic", cascade = {CascadeType.ALL})
    private List<Post> posts = new ArrayList<>();
    /**
     * Relation many to one avec les rubrique <br/>
     * Id dans table topic = rubrique_id <br/>
     * Voir {@link dev.entite.forum.Rubrique}
     */
    @ManyToOne
    private Rubrique rubrique;

    // Constructeur
    public Topic() {
    }

    // Getter & Setter
    public String getLibelle() {
        return libelle;
    }

    public void setLibelle(String libelle) {
        this.libelle = libelle;
    }

    public Utilisateur getUtilisateur() {
        return utilisateur;
    }

    public void setUtilisateur(Utilisateur utilisateur) {
        this.utilisateur = utilisateur;
    }

    public List<Post> getPosts() {
        return posts;
    }

    public void setPosts(List<Post> posts) {
        this.posts = posts;
    }

    public Rubrique getRubrique() {
        return rubrique;
    }

    public void setRubrique(Rubrique rubrique) {
        this.rubrique = rubrique;
    }
}

File: src/app/_shared/entities/Topic.ts
import {Rubrique} from "./Rubrique";
import {Utilisateur} from "./Utilisateur";
import {Post} from "./Post";

export class Topic {
  private _id: number
  private _libelle: string
  private _rubrique: Rubrique
  private _utilisateur: Utilisateur
  private _post: Post[]


  constructor(id: number, libelle: string, rubrique: Rubrique, Utilisateur: Utilisateur, post: Post[]) {
    this._id = id;
    this._libelle = libelle;
    this._rubrique = rubrique;
    this._utilisateur = Utilisateur;
    this._post = post;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get libelle(): string {
    return this._libelle;
  }

  set libelle(value: string) {
    this._libelle = value;
  }

  get rubrique(): Rubrique {
    return this._rubrique;
  }

  set rubrique(value: Rubrique) {
    this._rubrique = value;
  }

  get utilisateur(): Utilisateur {
    return this._utilisateur;
  }

  set utilisateur(value: Utilisateur) {
    this._utilisateur = value;
  }


  get post(): Post[] {
    return this._post;
  }

  set post(value: Post[]) {
    this._post = value;
  }
}
